### PR TITLE
feat: standalone "Add Liquidity"

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Wires-up "Add Liquidity" as a standalone page.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/6018

## Risk

Low - ensure everything behaves as expected. This is under feature-flag, however something something caution consider PR as if it was to land in prod

## Testing

- The "Add Liquidity" component looks sane both on click, and on page refresh
- It is possible to select any supported pool asset
- The component behaves the same as in the pool view except for two differences:
  - The pair selection is only possible when in standalone mode
  - The "Deposit Amounts" radio options are only displayed when in standalone mode
  
The reason for these two discrepancies is in a pool page view, you're already in the context of an asset + a/sym and can't change it

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #6028** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6028?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6028?utm_source=stack-comment).

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 1]
```diff
@@ -72,12 +74,55 @@ export const AddLiquidityInput: React.FC<AddLiquidityProps> = ({
...
+  // TODO(gomes): Even though that's an edge case for users, and a bad practice, handling sym and asymm positions simultaneously
+  // *is* possible and *is* something that both we and TS do. We can do one better than TS here however:
+  // - When a user deposits symetrically, they can then deposit asymetrically, but only on the asset side
+  // - When a user deposits asymetrically, no matter the side, they *can* deposit symetrically on the other side
+  //   - They can also deposit asymetrically after that, but with one caveat: they can do so only if they deposited asym on the *asset* side only
+  //     In other words, if they have an active asym. RUNE position, they can't deposit symetrically after that unless they withdraw
+  //     The reason for that is that the RUNE side memo performs a nameservice operation, registering the asset address (or a placeholder)
+  //
+  //     We should handle this in the UI and block users from deposits that *will* fail, by detecting their current position(s)
+  //     and not allowing them to select the sure-to-fail deposit types
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:93]: Yeah that's a long comment but we should absolutely handle this

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 2]
```diff
@@ -72,12 +74,55 @@ export const AddLiquidityInput: React.FC<AddLiquidityProps> = ({
...
+  const defaultOpportunityId = useMemo(() => {
     if (!parsedPools) return undefined
+    if (opportunityId) return undefined
 
-    return parsedPools.find(pool => pool.opportunityId === opportunityId)
+    // since we do asset/rune/sym ordering, we can assume the the third item is sym
+    return parsedPools[2].opportunityId
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:99]: We can do one better than this, this works for now but is not guaranteed in the future

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 3]
```diff
@@ -72,12 +74,55 @@ export const AddLiquidityInput: React.FC<AddLiquidityProps> = ({
...
   }, [opportunityId, parsedPools])
 
+  const [activeOpportunityId, setActiveOpportunityId] = useState(
+    opportunityId ?? defaultOpportunityId,
+  )
+
+  // TODO(gomes): find the *right* pool type depending on asym type when standalone
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:106]: Remove TODO

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 4]
```diff
@@ -72,12 +74,55 @@ export const AddLiquidityInput: React.FC<AddLiquidityProps> = ({
...
+
+  const _asset = useAppSelector(state => selectAssetById(state, foundPool?.assetId ?? ''))
+  const rune = useAppSelector(state => selectAssetById(state, thorchainAssetId))
+  const [asset, setAsset] = useState<Asset | undefined>(_asset)
+
+  useEffect(() => {
+    if (!(asset && parsedPools)) return
+
+    const filteredPools = parsedPools?.filter(pool => pool.assetId === asset.assetId)
+    // since we do asset/rune/sym ordering, we can assume the the third item is sym
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:121]: Ditto we can do one better than this, this works for now but is not guaranteed in the future

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 5]
```diff
@@ -284,6 +327,55 @@ export const AddLiquidityInput: React.FC<AddLiquidityProps> = ({
     )
   }, [asset, foundPool, rune, translate])
 
+  // TODO(gomes): obviously wrong
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:330]: Actually not wrong, but let's use the sell variant for declarativeness

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 6]
```diff
@@ -284,6 +327,55 @@ export const AddLiquidityInput: React.FC<AddLiquidityProps> = ({
...
+  const buyAssetSearch = useModal('buyAssetSearch')
+  const handlePoolAssetClick = useCallback(() => {
+    buyAssetSearch.open({
+      onClick: setAsset,
+      title: 'lending.borrow',
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:335]: no

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: cf7164938a

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx (hunk 3 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: 82e86fd8ba

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx (hunk 4 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: cf7164938a

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx (hunk 5 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: f379a195ff985ee6e72706421048ac9080e1696c

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 7]
```diff
@@ -284,43 +342,80 @@ export const AddLiquidityInput: React.FC<AddLiquidityProps> = ({
     )
   }, [asset, foundPool, rune, translate])
 
+  const buyAssetSearch = useModal('sellAssetSearch')
```
Comment (0xApotheosis) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:345]:
```suggestion
  const buyAssetSearch = useModal('buyAssetSearch')
```

**Review by 0xApotheosis — APPROVED** [review]
One non-blocking comment.

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx (hunk 7 above)
Comment (0xApotheosis) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: Fixed in https://github.com/shapeshift/web/pull/6028/commits/d943ba2003513a86a8d7066ad82213822a9db4e5

Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: Haha, I had actually used that one initiallly https://github.com/shapeshift/web/pull/6028#discussion_r1455487538 but decided sell made more sense!

**Pull request merged**